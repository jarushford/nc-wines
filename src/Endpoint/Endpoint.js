import React from 'react'

export default function Endpoint({ method, title, description, returned, required }) {
  return (
    <article className="endpoint">
      <h1>{method}  {title}</h1>
      <div className="endpt-info">
        <p>{description}</p>
        <h4>Returned Value: {returned}</h4>
        <h4>Required body for request: {required}</h4>
      </div>
    </article>
  )
}