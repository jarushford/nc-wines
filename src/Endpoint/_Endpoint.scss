.endpoint {

  h1 {
    font-family: 'Quicksand', sans-serif;
    font-size: 2.5rem;
    background: #8b2f6a;
    width: 75%;
    border-top-right-radius: 60px;
    border-bottom-right-radius: 60px;
    padding: 15px;
    color: white;
    margin-bottom: 0;
    position: relative;
    z-index: 2;
    clear: both;
  }
}

.endpt-info {
  width: 80%;
  float: right;
  clear: both;
  height: 200px;
  background: rgba(101, 3, 67, 0.2);
  margin-top: -15px;
  margin-bottom: 50px;
  border-bottom-left-radius: 30px;
  position: relative;
  z-index: 1;
  padding: 20px 15px;

  p,
  h4 {
    font-family: 'Quicksand', sans-serif;
    font-size: 1.25rem;
    margin: 25px 15px;
    width: 95%;
  }
}