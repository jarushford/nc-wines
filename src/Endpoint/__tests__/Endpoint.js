import React from 'react'
import { shallow } from 'enzyme'
import Endpoint from '../Endpoint'

describe('Endpoint', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Endpoint
      method="GET" 
      title="/api/v1/vineyards/:id" 
      description="Returns a single vineyard."
      returned="Object"
      required="N/A" 
    />)
    expect(wrapper).toMatchSnapshot()
  })
})