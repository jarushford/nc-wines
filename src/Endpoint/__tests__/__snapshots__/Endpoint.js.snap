// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Endpoint should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Endpoint
    description="Returns a single vineyard."
    method="GET"
    required="N/A"
    returned="Object"
    title="/api/v1/vineyards/:id"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          GET
            
          /api/v1/vineyards/:id
        </h1>,
        <div
          className="endpt-info"
        >
          <p>
            Returns a single vineyard.
          </p>
          <h4>
            Returned Value: 
            Object
          </h4>
          <h4>
            Required body for request: 
            N/A
          </h4>
        </div>,
      ],
      "className": "endpoint",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "GET",
            "  ",
            "/api/v1/vineyards/:id",
          ],
        },
        "ref": null,
        "rendered": Array [
          "GET",
          "  ",
          "/api/v1/vineyards/:id",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Returns a single vineyard.
            </p>,
            <h4>
              Returned Value: 
              Object
            </h4>,
            <h4>
              Required body for request: 
              N/A
            </h4>,
          ],
          "className": "endpt-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Returns a single vineyard.",
            },
            "ref": null,
            "rendered": "Returns a single vineyard.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Returned Value: ",
                "Object",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Returned Value: ",
              "Object",
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Required body for request: ",
                "N/A",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Required body for request: ",
              "N/A",
            ],
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            GET
              
            /api/v1/vineyards/:id
          </h1>,
          <div
            className="endpt-info"
          >
            <p>
              Returns a single vineyard.
            </p>
            <h4>
              Returned Value: 
              Object
            </h4>
            <h4>
              Required body for request: 
              N/A
            </h4>
          </div>,
        ],
        "className": "endpoint",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "GET",
              "  ",
              "/api/v1/vineyards/:id",
            ],
          },
          "ref": null,
          "rendered": Array [
            "GET",
            "  ",
            "/api/v1/vineyards/:id",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Returns a single vineyard.
              </p>,
              <h4>
                Returned Value: 
                Object
              </h4>,
              <h4>
                Required body for request: 
                N/A
              </h4>,
            ],
            "className": "endpt-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Returns a single vineyard.",
              },
              "ref": null,
              "rendered": "Returns a single vineyard.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Returned Value: ",
                  "Object",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Returned Value: ",
                "Object",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Required body for request: ",
                  "N/A",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Required body for request: ",
                "N/A",
              ],
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
